packages/superset-ui-plugin-chart-table/src/Table.tsx

```diff
@@ -176,6 +182,18 @@ class TableVis extends React.PureComponent<InternalTableProps, TableState> {
       }
     });
 
+    const keys = dataToRender && dataToRender.length > 0 ? Object.keys(dataToRender[0].data) : [];
+    let calculatedWidth = 0;
+    keys.forEach(key => {
+      const maxLength = Math.max(...data.map(d => String(d.data[key]).length), key.length);
+      columnMetadata[key] = {
+        maxWidth: MAX_COLUMN_WIDTH,
+        width: maxLength * CHAR_WIDTH,
```

Comment:
Can use this cheap estimate of `CHAR_WIDTH` or 

```ts
import { getMultipleTextDimensions } from `@superset-ui/dimension`;

// can also specify style
const dimensions = getMultipleTextDimensions({ texts: data.map(d => String(d.data[key]).concat([key]) });
const columnWidth = Math.max(...dimensions.map(dim => dim.width));
```

Reply:
It will perform not as good as the heuristics but should give more accurate column width.

Reply:
Will try this method :)

Reply:
After trying out this method, we decided to use `CHAR_WIDTH` for faster computation, and later when we move the logic to `<DataTable/>`, we can re-evaluate the approach.